aggregator_core/src/datastore/test_util.rs

```diff
@@ -17,20 +17,30 @@ use sqlx::{
     Connection, PgConnection,
 };
 use std::{
+    env,
     path::PathBuf,
     str::FromStr,
     sync::{Arc, Weak},
+    thread::JoinHandle,
     time::Duration,
 };
-use testcontainers::{runners::AsyncRunner, ContainerAsync, ContainerRequest, ImageExt};
-use tokio::sync::Mutex;
+use testcontainers::{runners::AsyncRunner, ContainerRequest, ImageExt};
+use tokio::{
+    io::{AsyncBufRead, AsyncBufReadExt},
+    join,
+    sync::{
+        oneshot::{self, Sender},
+        Mutex,
+    },
+};
 use tokio_postgres::{connect, Config, NoTls};
 use tracing::trace;
 
 use super::SUPPORTED_SCHEMA_VERSIONS;
 
 struct EphemeralDatabase {
-    _db_container: ContainerAsync<Postgres>,
+    db_thread: Option<JoinHandle<()>>,
+    db_thread_shutdown: Option<Sender<()>>,
```

Comment:
Is there a better data structure for `.take()`'ing something in a `Drop`? Option works fine enough, but maybe there's a more pertinent structure I'm missing?